src/drivers/amd/rembrandt/acp_sw_audio_dai.c

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+//Copyright(c) 2023 AMD. All rights reserved.
+//
+//Author:       Basavaraj Hiregoudar <basavaraj.hiregoudar@amd.com>
+//              Maruthi Machani <maruthi.machani@amd.com>
+
+#include <sof/drivers/acp_dai_dma.h>
+
+/* 8f00c3bb-e835-4767-9a34-b8ec1041e56b */
+DECLARE_SOF_UUID("swaudiodai", swaudiodai_uuid, 0x8f00c3bb, 0xe835, 0x4767,
+		 0x9a, 0x34, 0xb8, 0xec, 0x10, 0x41, 0xe5, 0x6b);
+DECLARE_TR_CTX(swaudiodai_tr, SOF_UUID(swaudiodai_uuid), LOG_LEVEL_INFO);
+
+static inline int swaudiodai_set_config(struct dai *dai, struct ipc_config_dai *common_config,
+					const void *spec_config)
+{
+	const struct sof_ipc_dai_config *config = spec_config;
+	struct acp_pdata *acpdata = dai_get_drvdata(dai);
+
+	acpdata->config = *config;
+	acpdata->sdw_params = config->acpsdw;
+
+	return 0;
+}
+
+static int swaudiodai_trigger(struct dai *dai, int cmd, int direction)
+{
+	/* nothing to do on rembrandt for SW dai */
+	return 0;
+}
+
+static int swaudiodai_probe(struct dai *dai)
+{
+	struct acp_pdata *acp;
+
+	dai_info(dai, "#$AMD$# SW dai probe");
+	/* allocate private data */
+	acp = rzalloc(SOF_MEM_ZONE_RUNTIME_SHARED, 0, SOF_MEM_CAPS_RAM, sizeof(*acp));
+	if (!acp) {
+		dai_err(dai, "SW dai probe alloc failed");
+		return -ENOMEM;
+	}
+	dai_set_drvdata(dai, acp);
+	return 0;
```

Comment:
Minor: to align with coding style, there should be one empty line before return:
```
    dai_set_drvdata(dai, acp);

    return 0;
}
```

Reply:
done